modules/virtual_network_gateway/variables.tf

```diff
@@ -0,0 +1,324 @@
+  description = "Should private IP be enabled on this gateway for connections?"
+  type        = bool
+}
+
+variable "ip_configuration" {
+  description = <<-EOF
+  List of IP configurations - every object in the list contains attributes:
+
+  - name - name of the IP configuration
+  - create_public_ip - boolean value, true if public IP needs to be created
+  - public_ip_name - name of the public IP resource used, when there is no need to create new one
+  - private_ip_address_allocation - defines how the private IP address of the gateways virtual interface is assigned. Valid options are Static or Dynamic. Defaults to Dynamic.
+  - public_ip_standard_sku - defaults to `false`, when set to `true` creates a Standard SKU, statically allocated public IP, otherwise it will be a Basic/Dynamic one.
+  - subnet_id - the ID of the gateway subnet of a virtual network in which the virtual network gateway will be created.
+
+  Example:
+
+  ip_configuration = [
+    {
+      name             = "001"
+      create_public_ip = true
+      subnet_id        = "ID_for_subnet_GatewaySubnet"
+    },
+    {
+      name             = "002"
+      create_public_ip = true
+      subnet_id        = "ID_for_subnet_GatewaySubnet"
+    }
+  ]
+
+  EOF
+  type        = list(any)
+}
+
+variable "vpn_client_configuration" {
+  description = <<-EOF
+  List of VPN client configurations - every object in the list contains attributes:
+  - address_space - the address space out of which IP addresses for vpn clients will be taken. You can provide more than one address space, e.g. in CIDR notation.
+  - aad_tenant - AzureAD Tenant URL
+  - aad_audience - the client id of the Azure VPN application. See Create an Active Directory (AD) tenant for P2S OpenVPN protocol connections for values
+  - aad_issuer - the STS url for your tenant
+  - root_certificate - one or more root_certificate blocks which are defined below. These root certificates are used to sign the client certificate used by the VPN clients to connect to the gateway.
+  - revoked_certificate - one or more revoked_certificate blocks which are defined below.
+  - radius_server_address - the address of the Radius server.
+  - radius_server_secret - the secret used by the Radius server.
+  - vpn_client_protocols - list of the protocols supported by the vpn client. The supported values are SSTP, IkeV2 and OpenVPN. Values SSTP and IkeV2 are incompatible with the use of aad_tenant, aad_audience and aad_issuer.
+  - vpn_auth_types - list of the vpn authentication types for the virtual network gateway. The supported values are AAD, Radius and Certificate.
+
+  EOF
+  type        = list(any)
+}
+
+variable "azure_bgp_peers_addresses" {
+  description = <<-EOF
+  Map of IP addresses used on Azure side for BGP. Map is used to not to duplicate IP address and refer to keys while configuring:
+  - custom_bgp_addresses
+  - peering_addresses in local_bgp_settings
+
+  Example:
+
+  azure_bgp_peers_addresses = {
```

Comment:
same suggestion as [here](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/pull/306#discussion_r1321159362)

Reply:
same reply as [here](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/pull/306#discussion_r1321333921)

Reply:
I've added checking number of Azure BGP APIPA IP addresses:

https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/872dce18522fce79418298c412c036ccf8e34c7b/modules/virtual_network_gateway/main.tf#L90-L93